addons/project_todo/static/tests/todo_done_checkmark.test.js

```diff
@@ -0,0 +1,202 @@
+                <field name="state" invisible="1"/>
+            </form>`,
+        search: `
+            <search/>`,
+        kanban: `
+            <kanban>
+                <template>
+                    <t t-name="kanban-box">
+                        <div>
+                            <field name="state" widget="todo_done_checkmark"/>
+                        </div>
+                    </t>
+                </template>
+            </kanban>
+        `,
+    };
+});
+
+test("Check clicking on todo_done_checkmark in kanban view with initial state is in progress", async () => {
+    onRpc("web_save", ({ args }) => {
+        expect(args[1].state).toBe("1_done", {
+            message: "The task should be marked as done",
+        });
+        expect.step("web_save");
+    });
+
+    await mountView({
+        resModel: "project.task",
+        type: "kanban",
+    });
+
+    const task1 = queryFirst(".o_kanban_record .o_todo_done_button");
+    expect(task1).not.toHaveClass("done_button_enabled", {
+        message: "The checkmark on this task should be displayed as undone as its state is 01_in_progress",
+    });
+    await task1.click();
+    await animationFrame();
+    expect(task1).toHaveClass("done_button_enabled", {
+        message: "The checkmark on this task should be displayed as done as its state is 1_done",
+    });
+    expect.verifySteps(["web_save"]);
+});
+
+test("Check clicking on todo_done_checkmark in kanban view with initial state is done", async () => {
+    onRpc("web_save", ({ args }) => {
+        expect(args[1].state).toBe("01_in_progress", {
+            message: "The task should be in progress",
+        });
+        expect.step("web_save");
+    });
+
+    await mountView({
+        resModel: "project.task",
+        type: "kanban",
+    });
+
+    const task2 = queryAll(".o_kanban_record .o_todo_done_button")[1];
+    expect(task2).toHaveClass("done_button_enabled", {
+        message: "The checkmark on this task should be displayed as done as its state is 1_done",
+    });
+    await task2.click();
+    await animationFrame();
+    expect(task2).not.toHaveClass("done_button_enabled", {
+        message: "The checkmark on this task should be displayed as undone as its state is 01_in_progress",
+    });
+    expect.verifySteps(["web_save"]);
+});
+
+test("Check clicking on todo_done_checkmark in list view with initial state is in progress", async () => {
+    onRpc("web_save", ({ args }) => {
+        expect(args[1].state).toBe("1_done", {
+            message: "The task should be marked as done",
+        });
+        expect.step("web_save");
+    });
+
+    await mountView({
+        resModel: "project.task",
+        type: "list",
+    });
+
+    const task1 = queryFirst(".o_todo_done_button");
+    expect(task1).not.toHaveClass("done_button_enabled", {
+        message: "The checkmark on this task should be displayed as undone as its state is 01_in_progress",
+    });
+    await task1.click();
```

Comment:
I don't think the click is asynchrone. 🤔 
if not, then await is not needed here
```suggestion
   task1.click();
```